wolfcrypt/src/ascon.c

```diff
@@ -0,0 +1,528 @@
+    a->s64[4] =
+         tmp.s64[4] ^ rotrFixed64(tmp.s64[4],  7) ^ rotrFixed64(tmp.s64[4], 41);
+    a->s64[1] =
+         tmp.s64[1] ^ rotrFixed64(tmp.s64[1], 61) ^ rotrFixed64(tmp.s64[1], 39);
+    a->s64[3] =
+         tmp.s64[3] ^ rotrFixed64(tmp.s64[3], 10) ^ rotrFixed64(tmp.s64[3], 17);
+    a->s64[0] =
+         tmp.s64[0] ^ rotrFixed64(tmp.s64[0], 19) ^ rotrFixed64(tmp.s64[0], 28);
+    a->s64[2] =
+         tmp.s64[2] ^ rotrFixed64(tmp.s64[2],  1) ^ rotrFixed64(tmp.s64[2],  6);
+}
+
+static void permutation(AsconState* a, byte rounds)
+{
+    byte i = start_index(rounds);
+    for (; i < MAX_ROUNDS; i++) {
+        ascon_round(a, i);
+    }
+}
+
+#else
+
+#define p(a, c) do {                                                           \
+    AsconState tmp;                                                            \
+    /* 2.6.1 Addition of Constants */                                          \
+    (a)->s64[2] ^= c;                                                          \
+    /* 2.6.2 Substitution Layer */                                             \
+    (a)->s64[0] ^= (a)->s64[4];                                                \
+    (a)->s64[4] ^= (a)->s64[3];                                                \
+    (a)->s64[2] ^= (a)->s64[1];                                                \
+    tmp.s64[0] = (a)->s64[0] ^ (~(a)->s64[1] & (a)->s64[2]);                   \
+    tmp.s64[2] = (a)->s64[2] ^ (~(a)->s64[3] & (a)->s64[4]);                   \
+    tmp.s64[4] = (a)->s64[4] ^ (~(a)->s64[0] & (a)->s64[1]);                   \
+    tmp.s64[1] = (a)->s64[1] ^ (~(a)->s64[2] & (a)->s64[3]);                   \
+    tmp.s64[3] = (a)->s64[3] ^ (~(a)->s64[4] & (a)->s64[0]);                   \
+    tmp.s64[1] ^= tmp.s64[0];                                                  \
+    tmp.s64[3] ^= tmp.s64[2];                                                  \
+    tmp.s64[0] ^= tmp.s64[4];                                                  \
+    tmp.s64[2] = ~tmp.s64[2];                                                  \
+    /* 2.6.3 Linear Diffusion Layer */                                         \
+    (a)->s64[4] =                                                              \
+       tmp.s64[4] ^ rotrFixed64(tmp.s64[4],  7) ^ rotrFixed64(tmp.s64[4], 41); \
+    (a)->s64[1] =                                                              \
+       tmp.s64[1] ^ rotrFixed64(tmp.s64[1], 61) ^ rotrFixed64(tmp.s64[1], 39); \
+    (a)->s64[3] =                                                              \
+       tmp.s64[3] ^ rotrFixed64(tmp.s64[3], 10) ^ rotrFixed64(tmp.s64[3], 17); \
+    (a)->s64[0] =                                                              \
+       tmp.s64[0] ^ rotrFixed64(tmp.s64[0], 19) ^ rotrFixed64(tmp.s64[0], 28); \
+    (a)->s64[2] =                                                              \
+       tmp.s64[2] ^ rotrFixed64(tmp.s64[2],  1) ^ rotrFixed64(tmp.s64[2],  6); \
+    } while (0)
```

Comment:
Pull while back an indent level or move everything in the block up another indent level.

Reply:
Pulling `while` back.